BDArmory/Control/MissileFire.cs

```diff
@@ -6582,16 +6573,7 @@ ml.TargetingMode switch
                             }
                             MissileLauncher mlauncher = ml as MissileLauncher;
 
-                            unguidedWeapon = ml.GuidanceMode == MissileBase.GuidanceModes.None ||
-                                ml.TargetingMode switch
-                                {
-                                    MissileBase.TargetingModes.None => true,
-                                    MissileBase.TargetingModes.Laser => BDATargetManager.ActiveLasers.Count <= 0,
-                                    MissileBase.TargetingModes.Radar => !_sonarsEnabled && (!ml.radarLOAL || (mlauncher != null && ml.radarTimeout < ((distanceToTarget - ml.activeRadarRange) / mlauncher.optimumAirspeed))),
-                                    MissileBase.TargetingModes.Inertial => !(_sonarsEnabled || _irstsEnabled),
-                                    MissileBase.TargetingModes.Gps => BDATargetManager.ActiveLasers.Count <= 0 && !_sonarsEnabled,
-                                    _ => false
-                                }; //unify unguidedWeapon conditions
+                            unguidedWeapon = ml.GuidanceMode == MissileBase.GuidanceModes.None || UnguidedMissile(ml, distanceToTarget);
```

Comment:
Leaving this as GuidanceModes.None || (various missing sensor conditions) means that deliberately setup dumb missiles like the AIR-2 are still going to be regarded by the AI things that can only be fired within 1/10th their maxEngageRange. if you want this to just be dumbfiring smart munitions then the GuidanceMode.none option should be removed from the bool setter